internal/controller/warehouses/images.go

```diff
@@ -130,10 +130,20 @@ func imageDiscoveryLogFields(sub kargoapi.ImageSubscription) []any {
 	}
 	switch sub.ImageSelectionStrategy {
 	case kargoapi.ImageSelectionStrategySemVer, kargoapi.ImageSelectionStrategyDigest:
-		f = append(
-			f,
-			"semverConstraint", sub.SemverConstraint,
-		)
+		// if the constraint field has been used, append its logs
+		// otherwise fallback to appending deprecated field logs
+		// only if semVer strategy is used
+		if sub.Constraint != "" {
+			f = append(
+				f,
+				"Constraint", sub.Constraint,
+			)
+		} else if sub.SemverConstraint != "" && sub.ImageSelectionStrategy == kargoapi.ImageSelectionStrategySemVer {
+			f = append(
+				f,
+				"semverConstraint", sub.SemverConstraint,
+			)
+		}
```

Comment:
The code would be much easier to follow if you replaced this conditional by splitting the switch statement's current single case for `ImageSelectionStrategySemVer` and `ImageSelectionStrategyDigest` into two separate cases.